### ISSUE TITLE
Make incoming port 25 check compatible with Azure

### ISSUE THREAD

**Issue by heinrich-ulbricht** [mail-in-a-box/mailinabox#1794]
I set up Mail-in-a-Box in a Azure VM which generally speaking worked well. But the self check tells me that _incoming_ port 25 is blocked:

`Incoming Mail (SMTP/postfix) is running but is not publicly accessible at xx.xx.xx.xx:25.`

This is not true. I confirmed that port 25 is open and reachable from the outside. 

This had me confused and even reach out to Azure support. But the problem seems to be that the self check tries to connect to itself on port 25 (that's an _outgoing_ connection) to test that _incoming_ port 25 is working. Azure blocks _outgoing_ connections to port 25 by default ([documentation](https://docs.microsoft.com/en-us/azure/virtual-network/troubleshoot-outbound-smtp-connectivity)).

If I understand this right then a proposed solution would be to have the incoming port 25 checked by a kind of proxy connecting from outside Azure. This should make the _incoming_ check work (which I expected in the first place). But before thinking too much about solutions I'd like to get other opinions on this issue.

**Comment by alento-group** [comment]
> Azure blocks outgoing connections to port 25 by default 

You need BOTH to run a mail server properly without the use of a SMTP relay.

**Comment by ddavness** [comment]
If i recall correctly from my brief usage of MIAB in Azure, even if outgoing port 25 is blocked, the fact that incoming port 25 is NOT blocked means we can still receive mail (for whatever purposes a receive-only email might be of use). So it's only fair if we use something else to test the incoming port 25 - on the other hand, it's not that great of an idea to rely too much on external services to do our testing as they can disappear overnight.

**Comment by heinrich-ulbricht** [comment]
Yes I tested this and can confirm MiaB can receive emails. And for my use case this is all I need, no outgoing connection necessary.

@ddavness The more I think about it the more I have to agree - making oneself dependent on an external service might not be the best idea, especially security-wise. A central service would know all MiaB instances. Not so great.

Maybe an alternative would be to adjust the self check. If outgoing port 25 fails then a warning (instead of error) could be shown that incoming port 25 cannot be checked. Right now the error message is plain wrong.
